packages/types/src/primitive/Method.ts

```diff
@@ -222,13 +310,17 @@ export default class Method extends Struct implements IMethod {
 
   /**
    * @description Returns the name of the method
+   *
+   * @deprecated Will be removed soon.
```

Comment:
tbh I think there's no need to deprecate this.

Phase 1 (now): we warn users that global inject will be gone soon, so that they do `new Method(value, meta)` in their code, and we extract methodName/sectionName from meta
Phase 2: we remove global inject, and make meta in constructor obligatory. We can then still extract methodName, sectionName.

Reply:
If we pass a FunctionMetadataV5 to the constructor, then we don't have the section name

Reply:
Agreed. This is why I would advocate to pass in whole metadata always.

Reply:
in my mind it somehow made sense to have the section and method name stored in MethodFunction, a wrapper around FunctionMetadata; and if we need to know/use the section/method name, then we use a MethodFunction

Reply:
however, considering the section/method name will have been passed to the constructor via meta if the developer wants to access it, I think it's safe to keep the getters :+1: